spec/03-plugins/03-http-log/01-log_spec.lua

```diff
@@ -269,4 +312,67 @@ for _, strategy in helpers.each_strategy() do
       end, 10)
     end)
   end)
-end
+
+  describe("Plugin: http-log (log) [#" .. strategy .. "]", function()
+    local proxy_client
+
+    setup(function()
+      local bp = helpers.get_db_utils(strategy)
+
+      bp.plugins:insert {
+        name     = "http-log",
+        config   = {
+          http_endpoint = "http://" .. helpers.mock_upstream_host
+                                    .. ":"
+                                    .. helpers.mock_upstream_port
+                                    .. "/post_log/http"
+        }
+      }
+
+      assert(helpers.start_kong({
+        database = strategy,
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+      }))
+    end)
+
+    teardown(function()
+      helpers.stop_kong()
+    end)
+
+    before_each(function()
+      proxy_client = helpers.proxy_client()
+    end)
+
+    after_each(function()
+      if proxy_client then
+        proxy_client:close()
+      end
+    end)
+
+    it("successfully executes globally when route does not exist", function()
+    	-- setup: cleanup logs
+      local test_error_log_path = helpers.test_conf.nginx_err_logs
+      os.execute(":> " .. test_error_log_path)
+
+      local res = assert(proxy_client:send({
+        method  = "GET",
+        path    = "/nonexistant/proxy/path",
+        headers = {
+          ["Host"] = "https_no_exist.test"
+        }
+      }))
+      assert.res_status(404, res)
+
+      -- Assertion: there should be no [error], including no error
+      -- resulting from attempting to reference the id on
+      -- a route or api when no such value exists
+
+      local pl_file = require "pl.file"
+      local logs = pl_file.read(test_error_log_path)
+
+      for line in logs:gmatch("[^\r\n]+") do
+        assert.not_match("[error]", line, nil, true)
+      end
```

Comment:
It seems like this test is not asserting that the plugin actually ran? Only that no error was thrown

Reply:
Agreed, whats the best way to validate a global plugin executed? I just sort of assumed it ran 😆, but certainly makes sense to validate that is the case.

Reply:
By copying any of the existing sanity tests in this suite (like “logs to http”) :)

Reply:
Attempt made if I am reading the code right. Thanks for the patience! Just a newb wanting to help make Kong great 👍 . Don't work too hard this Thanksgiving yo 😃 .